third_party/penumbra/src/gl/context.cpp

```diff
@@ -285,9 +285,11 @@ float Context::setScene(mat4x4 sunView, const SurfaceBuffer &surfaceBuffer, bool
   near_ = -MAX_FLOAT;
   far_ = MAX_FLOAT;
 
-  for (GLuint i = surfaceBuffer.begin * model.vertexSize;
-     i < surfaceBuffer.begin * model.vertexSize + surfaceBuffer.count * model.vertexSize;
-     i += model.vertexSize) {
+  // If surface buffer has not been set use entire model instead.
+  GLuint beg = surfaceBuffer ? surfaceBuffer->begin * model.vertexSize : 0;
```

Comment:
I'm sure there is some sentinel signal coming through to indicate that surfaceBuffer won't be accessed (`modelSet`?), but seeing a pointer with a default value of `nullptr` being accessed without a check makes me feel things.

Reply:
@Myoldmopar It looks like to me it is doing a pointer check, just by using a ternary instead of an if statement.

Reply:
Sorry, I saw the begin operator and didn't see right before it.  Thanks!